src/extension.ts

```diff
@@ -1,27 +1,44 @@
 import * as vscode from "vscode";
 import * as lc from "vscode-languageclient/node";
-import { setupLanguageServer } from "./cairols";
 import { Context } from "./context";
-
-let client: lc.LanguageClient | undefined;
+import { CairoExtensionManager } from "./extensionManager";
 
 export async function activate(extensionContext: vscode.ExtensionContext) {
   const ctx = Context.create(extensionContext);
 
   if (ctx.config.get("enableLanguageServer")) {
-    client = await setupLanguageServer(ctx);
+    const extensionManager = CairoExtensionManager.fromContext(ctx);
+    extensionManager.tryStartClient();
+
+    // Notify the server when the client configuration changes.
+    // CairoLS pulls configuration properties it is interested in by itself, so it
+    // is unnecessary to attach any details in the notification payload.
+    ctx.extension.subscriptions.push(
+      vscode.workspace.onDidChangeConfiguration(
+        async () => {
+          await extensionManager
+            .getClient()
+            ?.sendNotification(lc.DidChangeConfigurationNotification.type, {
+              settings: "",
```

Comment:
Can we use `null` here instead of string?
```suggestion
              settings: null,
```

Reply:
It's out of scope because it's only moved around

Reply:
ok